sklearn/model_selection/tests/test_split.py

```diff
@@ -0,0 +1,717 @@
+        assert_equal(np.sum(train == 1), np.sum(train == 2))
+
+
+def train_test_split_pandas():
+    # check train_test_split doesn't destroy pandas dataframe
+    types = [MockDataFrame]
+    try:
+        from pandas import DataFrame
+        types.append(DataFrame)
+    except ImportError:
+        pass
+    for InputFeatureType in types:
+        # X dataframe
+        X_df = InputFeatureType(X)
+        X_train, X_test = train_test_split(X_df)
+        assert_true(isinstance(X_train, InputFeatureType))
+        assert_true(isinstance(X_test, InputFeatureType))
+
+
+def train_test_split_mock_pandas():
+    # X mock dataframe
+    X_df = MockDataFrame(X)
+    X_train, X_test = train_test_split(X_df)
+    assert_true(isinstance(X_train, MockDataFrame))
+    assert_true(isinstance(X_test, MockDataFrame))
+    X_train_arr, X_test_arr = train_test_split(X_df)
+
+
+def test_shufflesplit_errors():
+    assert_raises(ValueError, ShuffleSplit, test_size=2.0)
+    assert_raises(ValueError, ShuffleSplit, test_size=1.0)
+    assert_raises(ValueError, ShuffleSplit, test_size=0.1, train_size=0.95)
+    assert_raises(ValueError, next, ShuffleSplit(test_size=11).split(X))
+    assert_raises(ValueError, next, ShuffleSplit(test_size=10).split(X))
+    assert_raises(ValueError, next, ShuffleSplit(test_size=8,
+                                                 train_size=3).split(X))
+    assert_raises(ValueError, ShuffleSplit, test_size=None, train_size=None)
+    assert_raises(ValueError, next, ShuffleSplit(train_size=1j).split(X))
+
+
+def test_shufflesplit_reproducible():
+    # Check that iterating twice on the ShuffleSplit gives the same
+    # sequence of train-test when the random_state is given
+    ss = ShuffleSplit(random_state=21)
+    assert_array_equal(list(a for a, b in ss.split(X)),
+                       list(a for a, b in ss.split(X)))
+
+
+def test_safe_split_with_precomputed_kernel():
+    clf = SVC()
+    clfp = SVC(kernel="precomputed")
+
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    K = np.dot(X, X.T)
+
+    cv = ShuffleSplit(test_size=0.25, random_state=0)
+    tr, te = list(cv.split(X))[0]
+
+    X_tr, y_tr = _safe_split(clf, X, y, tr)
+    K_tr, y_tr2 = _safe_split(clfp, K, y, tr)
+    assert_array_almost_equal(K_tr, np.dot(X_tr, X_tr.T))
+
+    X_te, y_te = _safe_split(clf, X, y, te, tr)
+    K_te, y_te2 = _safe_split(clfp, K, y, te, tr)
+    assert_array_almost_equal(K_te, np.dot(X_te, X_tr.T))
+
+
+def test_train_test_split_allow_nans():
+    # Check that train_test_split allows input data with NaNs
+    X = np.arange(200, dtype=np.float64).reshape(10, -1)
+    X[2, :] = np.nan
+    y = np.repeat([0, 1], X.shape[0] / 2)
+    train_test_split(X, y, test_size=0.2, random_state=42)
+
+
+def test_check_cv_return_types():
+    X = np.ones((9, 2))
+    cv = check_cv(3, classifier=False)
+    # Use numpy.testing.assert_equal which recursively compares
```

Comment:
It also checks if n_folds is set to 3 properly... (this could just be split into two test - check if it is a KFold instance and later check if `n_folds` is set to 3)